### ISSUE TITLE
add --verbose flag to helm test

### ISSUE THREAD

**Issue by michelleN** [helm/helm#1957]
When set to true, this flag will stream the test pod logs in the output

cc/ As per our discussion @foxish @erictune @vdice

**Comment by technosophos** [comment]
Would this be different in kind to supporting the already global `--debug` flag?

**Comment by technosophos** [comment]
@michelleN if this should go in 2.3.0, feel free to add it. For now, I'll drop it into triage.

**Comment by sstarcher** [comment]
This would be a nice addition.  Currently if you run `helm test --cleanup` and the job fails it will output.  ```FAILED: test, run `kubectl logs test --namespace test` for more info```

Which is not that useful as helm has already deleted the pod :)

**Comment by chrislovecnm** [comment]
What is the status on this item?

**Comment by lainosantos** [comment]
+1

**Comment by michelleN** [comment]
@chrislovecnm @lainosantos This kind of dropped off my radar as I got busy with other things. I can try to get on this in the next month or two.

**Comment by LSmith130** [comment]
Is this being worked on? I'm working on ci/cd scripts using helm, and I'm wondering if I should wait for this to be finished or try to parse helm test outputs and manually attach to the running tests. I would write the PR myself, but I'm not super comfortable with go and don't have time to pick it up right now.

**Comment by bacongobbler** [comment]
Nobody's currently working on this feature AFAIK. Sorry!

**Comment by tomstockton** [comment]
Here's my Bash workaround in the absence of this feature:
```
#!/bin/bash

helm_release=$1

test_pods=$(helm status $helm_release -o json | jq -r .info.status.last_test_suite_run.results[].name)
namespace=$(helm status $helm_release -o json | jq -r .namespace)

for test_pod in $test_pods; do
  echo "Test Pod: $test_pod"
  echo "============"
  echo ""
  kubectl -n $namespace logs $test_pod
  kubectl -n $namespace delete pod $test_pod
  echo ""
  echo "============"
done
```

**Comment by aelbarkani** [comment]
@michelleN any update on this feature ?

**Comment by michelleN** [comment]
@aelbarkani it looks like there is an open PR (#3907). It has been inactive for a few months but I just reached out to the developer to see if they're still interested in working on this functionality.

**Comment by markmssd** [comment]
I've used @tomstockton to write a bash file that will follow the logs as it's testing:
test.sh
```
# Define the Namespace
NAMESPACE=mynamespace

HELM_RELEASE=$NAMESPACE-123
POD_NAME=$HELM_RELEASE-test

echo "=====================
echo "Testing $HELM_RELEASE"
echo "=====================

# Delete previous test pod if any
kubectl delete pod $POD_NAME --ignore-not-found -n $NAMESPACE && sleep 5

# Keep checking the pod status every 5 seconds
# Stops when its phase is "Running"
# This allows us to start following logs only when "Running"
# "Unknown" happens if pod is not found
function is_pod_running() {
    UNKNOWN_COUNT=0
    while [[ ! $POD_STATE = "Running" ]]
    do
        sleep 5
        POD_STATE=$(kubectl get pod $POD_NAME --ignore-not-found -n $NAMESPACE -o=jsonpath='{.status.phase}')
        POD_STATE=${POD_STATE:-"Unknown"}

        if [[ $POD_STATE = "Unknown" ]]; then
            UNKNOWN_COUNT=$((UNKNOWN_COUNT + 1))
            if [[ $UNKNOWN_COUNT = 5 ]]; then
                echo "Pod in Unknown state for more than $UNKNOWN_COUNT times, cancelling"
                exit 1
            fi
        fi

        echo "$POD_NAME is $POD_STATE"
    done
}

# Follow logs
# Automatically stops once pod finishes executing (success or failure)
function follow_logs() {
    echo "Logs for $POD_NAME"
    echo "===================================="
    kubectl -n $NAMESPACE logs $POD_NAME --follow
}

# Run the tests
helm test $HELM_RELEASE & (is_pod_running && follow_logs)
```

**Comment by IRobL** [comment]
I really like how the verbose switch works in `kubectl`.  It probably makes development on kubernetes itself go a little bit faster because it's easier to see what's going on by adding `-v9` and be able to view all the REST requests being made against the cluster.  

Does anyone know off-hand what REST calls are made for `helm version`?  I'd like to be able to detect if tiller is installed in a namespace or not, and am having trouble sorting that out.

**Comment by bacongobbler** [comment]
Helm and Tiller communicate via gRPC, so it's not an HTTP request being made to Tiller, but the RPC call is listed [here](https://github.com/helm/helm/blob/master/_proto/hapi/version/version.proto).

**Comment by bacongobbler** [comment]
implemented in #6612
